AngularProject/ace/src/app/header/header/header.component.ts

```diff
@@ -1,15 +1,38 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { AuthService } from '../../auth/auth.service';
+import { CategoryService } from '../../sharedServices/category/category.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
 })
-export class HeaderComponent implements OnInit {
+export class HeaderComponent implements OnInit, OnDestroy {
+  public categories: Array<string>;
```

Comment:
same here.
You can store this as observable and pass it directly to the template using async pipe